### PR TITLE
Remove cached slack, client in App to reflect changes in AppConfig

### PR DESCRIPTION
**Pull request by seratch**
This pull request improves the internals of `App` class as below:

* `App#getSlack()` always returns a `Slack` instance with the latest configuration
* `App#getClient()` always returns a `MethodsClient` instance with the latest configuration
* Fix potential NPE error patterns

### Category (place an `x` in each of the `[ ]`)

* [x] **bolt** (Bolt for Java)
* [ ] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

### CODE REVIEW

File: bolt/src/main/java/com/slack/api/bolt/App.java [hunk 1]
```diff
@@ -70,12 +70,16 @@
     /**
      * Slack instance and its configurations to be used for API calls.
      */
-    private final Slack slack;
```
Comment (seratch) [inline bolt/src/main/java/com/slack/api/bolt/App.java:73]: Lombok's `@Data` annotation generates `getSlack()` method for this field.

File: bolt/src/main/java/com/slack/api/bolt/App.java [hunk 2]
```diff
@@ -441,8 +445,6 @@ public App(AppConfig appConfig, List<Middleware> middlewareList) {
 
     public App(AppConfig appConfig, Slack slack, List<Middleware> middlewareList) {
         this.appConfig = appConfig;
-        this.slack = slack;
-        this.client = this.slack.methods();
```
Comment (seratch) [inline bolt/src/main/java/com/slack/api/bolt/App.java:445]: Removed these cached instances. Accessing `config().getSlack()` instead has almost zero overhead.

File: slack-api-client/src/main/java/com/slack/api/methods/impl/TeamIdCache.java [hunk 3]
```diff
@@ -32,15 +32,16 @@ public String lookupOrResolve(String token) {
                 FormBody.Builder form = toForm(AuthTestRequest.builder().token(newToken).build());
                 Response response = methodsImpl.runPostFormWithToken(form, Methods.AUTH_TEST, token);
                 AuthTestResponse authTest = methodsImpl.parseJsonResponseAndRunListeners(null, null, response, AuthTestResponse.class);
-                if (authTest.isOk()) {
+                if (authTest != null && authTest.isOk()) {
```
Comment (seratch) [inline slack-api-client/src/main/java/com/slack/api/methods/impl/TeamIdCache.java:35]: While writing a mock API server and doing tests with it, I found the potential null pattern. Except the situation where the Slack server-side returns empty response body, `authTest` cannot be null, though.

File: slack-api-client/src/main/java/com/slack/api/methods/impl/TeamIdCache.java [hunk 4]
```diff
@@ -32,15 +32,16 @@ public String lookupOrResolve(String token) {
...
                     if (log.isDebugEnabled()) {
                         log.debug("Created cache for an auth.test API call (token: {}, team_id: {})",
                                 token.substring(0, 16) + "...", authTest.getTeamId());
                     }
                     // for org admin user's token, this value can be an enterprise_id
                     return authTest.getTeamId();
                 } else {
-                    log.error("Got an unsuccessful response from auth.test API (error: {})", authTest.getError());
+                    String error = authTest != null ? authTest.getError() : "";
```
Comment (filmaj) [inline slack-api-client/src/main/java/com/slack/api/methods/impl/TeamIdCache.java:43]: Perhaps in the case that `authTest` is null, we set a more descriptive error than the empty string? I know you said `authTest` cannot be null - but who knows how this code will change in the future 😃 

```suggestion
                    String error = authTest != null ? authTest.getError() : "AuthTestResponse is null";
```

**Review by filmaj — APPROVED** [review]
👍

File: slack-api-client/src/main/java/com/slack/api/methods/impl/TeamIdCache.java (hunk 4 above)
Comment (seratch) [reply slack-api-client/src/main/java/com/slack/api/methods/impl/TeamIdCache.java:43]:
```suggestion
                    String error = authTest != null ? authTest.getError() : "(empty response)";
```

Comment (seratch) [reply slack-api-client/src/main/java/com/slack/api/methods/impl/TeamIdCache.java:43]: Thanks! Perhaps, mentioning an empty response body would be easier to understand

**Pull request merged**